web-stories.php

```diff
@@ -60,15 +65,141 @@
 	define( 'WEBSTORIES_DEV_MODE', false );
 }
 
+// Load Compatibility class the old fashioned way.
+if ( ! class_exists( 'Web_Stories_Compatibility' ) ) {
+	require_once WEBSTORIES_PLUGIN_DIR_PATH . 'Web_Stories_Compatibility.php';
+}
+
+global $web_stories_compatibility;
+
+$web_stories_error = new WP_Error();
+$extensions        = array(
+	'date'   => array(
+		'classes' => array(
+			'DateTimeImmutable',
+		),
+	),
+	'dom'    => array(
+		'classes' => array(
+			'DOMAttr',
+			'DOMComment',
+			'DOMDocument',
+			'DOMElement',
+			'DOMNode',
+			'DOMNodeList',
+			'DOMText',
+			'DOMXPath',
+		),
+	),
+	'json'   => array(
+		'functions' => array(
+			'json_decode',
+			'json_encode',
+		),
+	),
+	'libxml' => array(
+		'functions' => array(
+			'libxml_use_internal_errors',
+		),
+	),
+	'spl'    => array(
+		'functions' => array(
+			'spl_autoload_register',
+		),
+	),
+);
+
+$web_stories_compatibility = new Web_Stories_Compatibility( $web_stories_error );
+$web_stories_compatibility->set_extensions( $extensions );
+$web_stories_compatibility->set_php_version( WEBSTORIES_MINIMUM_PHP_VERSION );
+$web_stories_compatibility->set_wp_version( WEBSTORIES_MINIMUM_WP_VERSION );
+$web_stories_compatibility->set_required_files(
+	array(
+		WEBSTORIES_PLUGIN_DIR_PATH . '/assets/js/edit-story.js',
+		WEBSTORIES_PLUGIN_DIR_PATH . '/assets/js/stories-dashboard.js',
+		WEBSTORIES_PLUGIN_DIR_PATH . '/assets/js/web-stories-embed-block.js',
+		WEBSTORIES_PLUGIN_DIR_PATH . '/includes/vendor/autoload.php',
+		WEBSTORIES_PLUGIN_DIR_PATH . '/third-party/vendor/scoper-autoload.php',
+	)
+);
+
+/**
+ * Displays an admin notice about why the plugin is unable to load.
+ *
+ * @since 1.0.0
+ *
+ * @return void
+ */
+function web_stories_print_admin_notice() {
```

Comment:
Now wondering... Could these functions simply be methods of the compatibility class?

Reply:
that doesnt seem right. The compatibility class feels like just the checks. These output are implementations of these checks. Maybe another class...